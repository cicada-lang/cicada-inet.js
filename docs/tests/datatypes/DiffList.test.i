require "DiffList.i"

claim one_two_soles -- Trivial DiffList end

define one_two_soles
  sole (cons :tail) diff rot rot connect
  sole (cons :tail) sole cons diff rot rot connect
  diff_append
end

begin one_two_soles inspect run inspect end

claim two_two_soles -- Trivial DiffList end

define two_two_soles
  sole (cons :tail) sole cons diff rot rot connect
  sole (cons :tail) sole cons diff rot rot connect
  diff_append
end

begin two_two_soles inspect run inspect end

claim two_two_soles_with_local -- Trivial DiffList end

define two_two_soles_with_local
  sole (cons :tail) sole cons diff $value connect value
  sole (cons :tail) sole cons diff $value connect value
  diff_append
end

begin two_two_soles_with_local inspect run inspect end
