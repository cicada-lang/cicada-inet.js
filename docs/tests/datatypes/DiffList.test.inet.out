(diff₆)-value!target-(diff_append₂)
(diff₅)-value rest-(diff_append₂)
(diff₆)-back tail-(cons₄)
(cons₅)-value front-(diff₆)
(cons₄)-value tail-(cons₅)
(sole₂)-value head-(cons₅)
(sole₁)-value head-(cons₄)
(diff₅)-back tail-(cons₃)
(cons₃)-value front-(diff₅)
(sole₀)-value head-(cons₃)
(diff_append₂)-return

(cons₃)-value tail-(cons₄)
(cons₃)-tail back-(diff₇)
(cons₅)-value front-(diff₇)
(cons₄)-value tail-(cons₅)
(sole₂)-value head-(cons₅)
(sole₁)-value head-(cons₄)
(sole₀)-value head-(cons₃)
(diff₇)-value

(diff₉)-value!target-(diff_append₃)
(diff₈)-value rest-(diff_append₃)
(diff₉)-back tail-(cons₈)
(cons₉)-value front-(diff₉)
(cons₈)-value tail-(cons₉)
(sole₆)-value head-(cons₉)
(sole₅)-value head-(cons₈)
(diff₈)-back tail-(cons₆)
(cons₇)-value front-(diff₈)
(cons₆)-value tail-(cons₇)
(sole₄)-value head-(cons₇)
(sole₃)-value head-(cons₆)
(diff_append₃)-return

(cons₇)-value tail-(cons₈)
(cons₆)-tail back-(diff₁₀)
(cons₉)-value front-(diff₁₀)
(cons₈)-value tail-(cons₉)
(sole₆)-value head-(cons₉)
(sole₅)-value head-(cons₈)
(cons₆)-value tail-(cons₇)
(sole₄)-value head-(cons₇)
(sole₃)-value head-(cons₆)
(diff₁₀)-value

(diff₁₂)-value!target-(diff_append₄)
(diff₁₁)-value rest-(diff_append₄)
(diff₁₂)-back tail-(cons₁₂)
(cons₁₃)-value front-(diff₁₂)
(cons₁₂)-value tail-(cons₁₃)
(sole₁₀)-value head-(cons₁₃)
(sole₉)-value head-(cons₁₂)
(diff₁₁)-back tail-(cons₁₀)
(cons₁₁)-value front-(diff₁₁)
(cons₁₀)-value tail-(cons₁₁)
(sole₈)-value head-(cons₁₁)
(sole₇)-value head-(cons₁₀)
(diff_append₄)-return

(cons₁₁)-value tail-(cons₁₂)
(cons₁₀)-tail back-(diff₁₃)
(cons₁₃)-value front-(diff₁₃)
(cons₁₂)-value tail-(cons₁₃)
(sole₁₀)-value head-(cons₁₃)
(sole₉)-value head-(cons₁₂)
(cons₁₀)-value tail-(cons₁₁)
(sole₈)-value head-(cons₁₁)
(sole₇)-value head-(cons₁₀)
(diff₁₃)-value

